spec/controllers/systems_spec.cr

```diff
@@ -362,58 +393,58 @@ module PlaceOS::Api
-        mod = Model::Generator.module(driver: driver).save!
-        module_id = mod.id.as(String)
+        # "fetches the state for `key` in module defined by `module_slug`
+        it "GET /:sys_id/:module_slug/:key" do
+          module_slug = cs.modules.first
 
-        cs.add_module(module_id)
-        cs.save!
+          # Create a storage proxy
+          driver_proxy = PlaceOS::Driver::RedisStorage.new mod.id.as(String)
 
-        module_slug = cs.modules.first
+          status_name = "orange"
+          driver_proxy[status_name] = 1
 
-        path = base + "#{cs.id}/functions/#{module_slug}"
+          sys_lookup = PlaceOS::Driver::RedisStorage.new(cs.id.as(String), "system")
+          lookup_key = "#{module_slug}/1"
+          sys_lookup[lookup_key] = mod.id.as(String)
 
-        result = curl(
-          method: "GET",
-          path: path,
-          headers: authorization_header,
-        )
+          path = base + "#{cs.id}/#{module_slug}/orange"
 
-        puts result.inspect
-      end
-
-      # TODO: awaiting addition of `core` to test environment
-      pending "GET /:sys_id/:module_slug/:key" do
-        it "fetches the state for `key` in module defined by `module_slug`" do
+          response = curl(
+            method: "GET",
+            path: path,
+            headers: authorization_header.merge({"Content-Type" => "application/json"}),
+          )
+          String.from_json(response.body).should eq("1")
         end
-      end
 
-      # TODO: awaiting addition of `core` to test environment
-      pending "GET /:sys_id/:module_slug" do
-        cs = Model::Generator.control_system.save!
+        it "GET /:sys_id/:module_slug" do
+          module_slug = cs.modules.first
 
-        driver = Model::Generator.driver(role: Model::Driver::Role::SSH).save!
-        mod = Model::Generator.module(driver: driver).save!
-        module_id = mod.id.as(String)
+          # Create a storage proxy
+          driver_proxy = PlaceOS::Driver::RedisStorage.new mod.id.as(String)
 
-        cs.add_module(module_id)
-        cs.save!
+          status_name = "nugget"
+          driver_proxy[status_name] = 1
 
-        module_slug = cs.modules.first
+          sys_lookup = PlaceOS::Driver::RedisStorage.new(cs.id.as(String), "system")
+          lookup_key = "#{module_slug}/1"
+          sys_lookup[lookup_key] = mod.id.as(String)
 
-        path = base + "#{cs.id}/functions/#{module_slug}"
+          path = base + "#{cs.id}/#{module_slug}"
 
-        result = curl(
-          method: "GET",
-          path: path,
-          headers: authorization_header,
-        )
+          response = curl(
+            method: "GET",
+            path: path,
+            headers: authorization_header.merge({"Content-Type" => "application/json"}),
+          )
 
-        puts result.inspect
+          state = Hash(String, String).from_json(response.body)
+          state["nugget"].should eq("1")
+        end
+        clear_tables
```

Comment:
What is the explicit `clear_tables` for? Is it to get core into the same default state?

Reply:
Yeah

Reply:
To make sure specs run as quick as possible (despite cloning/building drivers), try to get the set up to create or restore to a default state the same set of resources